src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -28,12 +47,17 @@ export const TodoFilter: React.FC = () => {
         </span>
 
         <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
+
+          {/* eslint-disable */}
+
+          {data.query && (
+            <button
+              data-cy="clearSearchButton"
+              onClick={() => dispatch(filterActions.removeQuery())}
```

Comment:
no need it separate action creator for reseting the query, just pass `''` to action creator that handle value updating